src/pages/index.js

```diff
@@ -95,13 +194,48 @@ addCardPopup.setEventListeners();
 previewImagePopup.setEventListeners();
 
 function handleAddCardFormSubmit(inputValues) {
-  const { name, link } = inputValues;
-  const card = renderCard({ name, link });
-  cardSection.addItem(card);
-  addCardPopup.close();
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCardInformation(inputValues.name, inputValues.link)
+    .then((res) => {
+      const newCard = renderCard(res);
+      cardSection.addItem(newCard);
+      addCardPopup.close();
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 addNewCardButton.addEventListener("click", () => {
   addCardValidator.toggleButtonState();
   addCardPopup.open();
 });
+
+profileAvatarButton.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.userName;
+  avatarValidator.toggleButtonState();
+  avatarInformation.open();
+});
+
+function handleAvatarImage(inputValues) {
+  avatarInformation.renderLoading(true);
+  api
+    .avatarInformation(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatarInfo(inputValues.avatar);
+    })
+    .then(() => {
+      avatarInformation.close();
+    })
+    .catch((err) => {
+      console.error(err);

```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors